google/cloud/bigtable/internal/legacy_row_reader_impl.cc

```diff
@@ -105,25 +105,28 @@ bool LegacyRowReaderImpl::NextChunk() {
   return true;
 }
 
-StatusOr<RowReaderImpl::OptionalRow> LegacyRowReaderImpl::Advance() {
+absl::variant<Status, bigtable::Row> LegacyRowReaderImpl::Advance() {
   if (operation_cancelled_) {
     return Status(StatusCode::kCancelled, "Operation cancelled.");
   }
   while (true) {
-    OptionalRow row;
-    grpc::Status status = AdvanceOrFail(row);
+    auto variant = AdvanceOrFail();
+    if (absl::holds_alternative<bigtable::Row>(variant)) {
+      return absl::get<bigtable::Row>(std::move(variant));
+    }
+
+    auto status = absl::get<Status>(std::move(variant));
     if (status.ok()) {
-      return row;
+      return Status();
```

Comment:
nit: nowadays we would write `if (status.ok()) return Status{};`

Reply:
Here and some places below.

Reply:
Fixed.